ts/components/ui/LogoPaymentExtended.tsx

```diff
@@ -6,9 +6,15 @@ import BpayLogo from "../../../img/wallet/payment-methods/bpay_logo_full.svg";
 import { BankLogoOrSkeleton } from "./utils/components/BankLogoOrLoadingSkeleton";
 export type LogoPaymentExtendedProps = {
   dimensions: { height: number; width: number };
-  icon?: "payPal" | "bpay";
-  abiCode?: string;
-};
+} & (
+  | {
+      icon: "payPal" | "bpay";
+    }
+  | {
+      abiCode: string | undefined;
+      imageA11yLabel?: string;
+    }
+);
```

Comment:
since `imageA11yLabel` has any use only in case an abiCode is passed, I think this is a cleaner solution.
After all it's 3 more lines of code and it's a bit easier to understand logically imho; what do you think?

Reply:
In my opinion it's not easier to understand and it's not logically correct. 
You can try replacing the code and you can see that nothing changes because they are equal.

```typescript
// This object has no type errors!
const test: LogoPaymentExtendedProps = {
  dimensions: { height: 100, width: 100 },
  icon: "payPal",
  abiCode: "123",
};
```

What you tried to do is something like this:

```typescript
export type LogoPaymentExtendedProps = {
  dimensions: { height: number; width: number };
} & (
  | {
      icon: "payPal" | "bpay";
    }
  | {
      icon: undefined;
      abiCode: string | undefined;
      imageA11yLabel?: string;
    }
);
```

This type tells me that I can make an object of type `LogoPaymentExtendedProps` with an `icon` OR an `abiCode`. 
Valid objects are:

```typescript
// No errors
const test: LogoPaymentExtendedProps = {
  dimensions: { height: 100, width: 100 },
  icon: "payPal",
};

// No errors
const test: LogoPaymentExtendedProps = {
  dimensions: { height: 100, width: 100 },
  abiCode: "123",
};

// Errors!
const a: LogoPaymentExtendedProps = {
  dimensions: { height: 100, width: 100 },
  icon: "payPal", <---
  abiCode: "123", <--- "this town's too small for both of us"
};

```

If you want to achieve the second case add `icon: undefined`,  remove the union otherwise.
I hope it's clear now! 😄

Reply:
right, updated as suggested 👍